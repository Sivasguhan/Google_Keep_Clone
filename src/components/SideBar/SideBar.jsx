import ImageButton from '../UI/ImageButton/ImageButton'
import Reload from '../../../assets/images/reload.png'
import Notes from '../../../assets/images/notes.jpeg'
import Remainder from '../../../assets/images/remainder.png'
import EditLabel from '../../../assets/images/editlabel.png'
import Archive from '../../../assets/images/archive.png'
import Trash from '../../../assets/images/trash.png'
import { useState } from 'react'

export default function SideBar({ open }) {

    const [isHovering, setIsHovering] = useState(false)

    const shouldOpen = open || isHovering;

    return (
        <nav
            className={open ? 'sidebar navOpenStyle' : 'sidebar'}
            onMouseEnter={() => setIsHovering(true)}
            onMouseLeave={() => setIsHovering(false)}
        >
            <div>
                <ImageButton imageUrl={Notes} />
                {shouldOpen && <span>Notes</span>}
            </div>
            <div>
                <ImageButton imageUrl={Remainder} />
                {shouldOpen && <span>Remainder</span>}
            </div>
            <div>
                <ImageButton imageUrl={EditLabel} />
                {shouldOpen && <span>Edit Labels</span>}
            </div>
            <div>
                <ImageButton imageUrl={Archive} />
                {shouldOpen && <span>Archive</span>}
            </div>
            <div>
                <ImageButton imageUrl={Trash} />
                {shouldOpen && <span>Trash</span>}
            </div>
        </nav>
    )
}