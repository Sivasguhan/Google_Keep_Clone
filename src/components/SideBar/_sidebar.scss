.sidebar {
    height: calc(100vh - 65px);
    width: 40px;
    background-color: white;
    transition: ease-out 0.4s;
    cursor: pointer;
    overflow: hidden;

    div {
        @include flex-style(row,flex-start,center);
        gap: 0.5rem;
        padding-top: 10px;

        &:hover {
            background-color: rgba(60, 64, 67, .08);
        }
    }

    &:hover {
        width: 250px;
        border-right: 1px solid $border-color;
    }

}

.navOpenStyle {
    width: 250px;
    border-right: 1px solid $border-color;
}