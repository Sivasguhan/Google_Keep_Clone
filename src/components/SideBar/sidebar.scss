.sidebar {
    height: calc(100vh - 65px);
    width: 40px;
    background-color: white;
    transition: ease-out 0.4s;
    cursor: pointer;
    overflow: hidden;

    div {
        display: flex;
        gap: 0.5rem;
        align-items: center;
        padding-top: 10px;

        &:hover {
            background-color: rgba(60, 64, 67, .08);
        }
    }

    &:hover {
        width: 250px;
        border-right: 1px solid rgb(209, 190, 190);
    }

}

.navOpenStyle {
    width: 250px;
    border-right: 1px solid rgb(209, 190, 190);
}