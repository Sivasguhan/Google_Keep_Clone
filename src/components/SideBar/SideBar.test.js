import { render } from "@testing-library/react";
import userEvent from '@testing-library/user-event';
import SideBar from "./SideBar";
import renderer from 'react-test-renderer'

test('initial state is set correctly', () => {
    const { queryByText } = render(<SideBar />);
    expect(queryByText('Notes')).toBeNull();
    expect(queryByText('Remainder')).toBeNull();
    expect(queryByText('Edit Labels')).toBeNull();
    expect(queryByText('Archive')).toBeNull();
    expect(queryByText('Trash')).toBeNull();
});

test('sidebar reacts to mouse hover', () => {
    const { container } = render(<SideBar open={true} />);
    const sidebar = container.querySelector('.sidebar');
    userEvent.hover(sidebar);
    expect(container.querySelectorAll('span')).toHaveLength(5);
});

test("Snapshot test", () => {
    const snap = renderer.create(<SideBar />).toJSON();
    expect(snap).toMatchSnapshot();
})