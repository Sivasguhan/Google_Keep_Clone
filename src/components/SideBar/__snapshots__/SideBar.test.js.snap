// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<nav
  className="sidebar"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div>
    <div
      className="imageContainer"
      style={
        {
          "height": "40px",
          "padding": "5px",
          "width": "40px",
        }
      }
    >
      <img
        src="mock-file"
      />
    </div>
  </div>
  <div>
    <div
      className="imageContainer"
      style={
        {
          "height": "40px",
          "padding": "5px",
          "width": "40px",
        }
      }
    >
      <img
        src="mock-file"
      />
    </div>
  </div>
  <div>
    <div
      className="imageContainer"
      style={
        {
          "height": "40px",
          "padding": "5px",
          "width": "40px",
        }
      }
    >
      <img
        src="mock-file"
      />
    </div>
  </div>
  <div>
    <div
      className="imageContainer"
      style={
        {
          "height": "40px",
          "padding": "5px",
          "width": "40px",
        }
      }
    >
      <img
        src="mock-file"
      />
    </div>
  </div>
  <div>
    <div
      className="imageContainer"
      style={
        {
          "height": "40px",
          "padding": "5px",
          "width": "40px",
        }
      }
    >
      <img
        src="mock-file"
      />
    </div>
  </div>
</nav>
`;
